server/chat-plugins/help.ts

```diff
@@ -0,0 +1,473 @@
+/**
+ * PS Help room auto-response plugin.
+ * Uses Regex to match room frequently asked question (RFAQ) entries,
+ * and replies if a match is found.
+ * Supports configuration.
+ * Written by mia-pi.
+ */
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {LogViewer} from './chatlog';
+import {ROOMFAQ_FILE} from './room-faqs';
+const PATH = 'config/chat-plugins/help.json';
+const MINIMUM_LENGTH = 7;
+const helpRoom = Rooms.get('help');
+const REGEX_WHITELIST = ['miapi', 'annika', 'kris'];
```

Comment:
@Zarel 

This is the main thing I would like to talk about. This is ugly AF in code we're checking in, in the same way it was ugly AF to check in the anubis hardcode in the datacenters code. I agree that adding a parallel whitelist command (and the necessarily persistence of the whitelist etc) is overkill, but adding a *general* whitelist that all commands can use is something I think is valuable:

`/whitelist helpfilter miapi` or `/whitelist datacenters anubis` etc. You then only need to set up the whitelisting infrastructure one and be able to apply it to all commands. Yes, this does further decouple the auth symbol from the powers that comes with it, but is clearly something that does come up quite often.

Reply:
I really really don't like whitelisting in general, but I guess if it's going to be done anyway...

In this specific case: what if we just used "roomdriver in Help and Dev" as the whitelist? Maybe the Annika whitelist could also just be "roommod in Dev".

Reply:
Roommod in Dev seems rather broad to be granting access to IP range and host management.

Reply:
I think there are very real reasons we might want to consider given Anubis access to more commands then just datacenters as well...

Reply:
Gmod + roommod in dev, then? I don't think it's too broad, I'd definitely trust anyone with that specific permission combination.

Reply:
I think if Anubis wants to be a special snowflake I will grant it based on all she's contributed to PS, but the hardcode also reflects my extreme distaste for the practice in general and is at least partially intended to discourage others from trying to get at that.

Reply:
Can you elaborate on your concerns over whitelisting? You're literally already allowing whitelists in at least 5 places that I'm aware of with hardcoded lists of users ids (datacenters, here, pmlogs, sysops, console)

Reply:
I suppose I don't see a problem with whitelisting specific admin users. I don't like whitelists below that, though, because the rank system should ideally already cover it.